File: src/main/java/io/gitlab/intended/storagenetworks/item/ModItem.java
package io.gitlab.intended.storagenetworks.item;

import io.gitlab.intended.storagenetworks.StorageNetworks;
import io.gitlab.intended.storagenetworks.registry.BlockRegistry;
import net.fabricmc.fabric.api.client.itemgroup.FabricItemGroupBuilder;
import net.minecraft.client.item.TooltipContext;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.item.ItemStack;
import net.minecraft.text.Text;
import net.minecraft.text.TranslatableText;
import net.minecraft.world.World;

import java.util.List;

public abstract class ModItem extends Item {

    public static final ItemGroup ITEM_GROUP = FabricItemGroupBuilder.build(StorageNetworks.id("item_group"), () -> new ItemStack(BlockRegistry.MASTER));

    public ModItem(Settings settings) {
        super(settings);
    }

    public static Settings getSettings() {
        return new Settings().group(ITEM_GROUP);
    }

    @Override
    public void appendTooltip(ItemStack stack, World world, List<Text> tooltip, TooltipContext context) {
        tooltip.add(new TranslatableText(stack.getTranslationKey() + ".tooltip"));
    }

}

File: src/main/kotlin/io/gitlab/intended/storagenetworks/item/ModItem.kt
package io.gitlab.intended.storagenetworks.item

import io.gitlab.intended.storagenetworks.Mod
import io.gitlab.intended.storagenetworks.block.ModBlocks
import net.fabricmc.fabric.api.client.itemgroup.FabricItemGroupBuilder
import net.minecraft.item.Item
import net.minecraft.item.ItemStack

abstract class ModItem(settings: Settings) : Item(settings) {

    companion object {
        private val GROUP = FabricItemGroupBuilder
            .create(Mod.id("group"))
            .icon { ItemStack(ModBlocks.MASTER) }
            .appendItems { stack ->
                ModBlocks.BLOCKS.forEach { block ->
                    stack.add(ItemStack(block))
                }
            }
            .build()

        val settings: Settings get() = Settings().group(GROUP)
    }

}
